docs/types/operators.rst

```diff
@@ -15,6 +15,14 @@ and the type of the operator's result:
    operand, use the type of the right operand,
 3. otherwise, the operation is not allowed.
 
+.. Warning::
```

Comment:
The [section about literals](https://docs.soliditylang.org/en/latest/types.html#rational-and-integer-literals) needs clarification too.

First, it says `by using them together with a non-literal expression` and to me this does not imply that `true` and `false` count too. They are not "non-literal expressions".

Second, I think this section needs a warning too, pointing out the behavior of the ternary operator. I'd mention the case of array literals too (i.e. something like `[255][0]` being of `uint8` type and not a literal expression).